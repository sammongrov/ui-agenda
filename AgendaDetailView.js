import React, { Component } from 'react';
import { View, TouchableOpacity } from 'react-native';
import PropTypes from 'prop-types';
import { Actions } from 'react-native-router-flux';
import { Colors } from '@ui/theme_default';
import { NavBar, Text, Screen, Icon, AntDesignIcon } from '@ui/components';
import { styles } from 'react-native-theme';
import {Application} from "@mongrov/config"; 

export default class AgendaDetailView extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
  };

  static defaultProps = {};

  state = {};

  render() {
    const { eventDetail } = this.props;
    return (
      <Screen>
        <NavBar
          titleText="Event Details"
          leftComponent={
            <TouchableOpacity
              onPress={() => {
                if (Actions.currentScene === 'AgendaDetailViewScene') {
                  Actions.pop();
                }
              }}
            >
              <Icon name="chevron-left" color={Colors.NAV_ICON} size={36} />
            </TouchableOpacity>
          }
          rightComponent={
            Application.APPCONFIG.FORM_ENABLE && (
              <TouchableOpacity
                style={{ marginRight: 15 }}
                onPress={() => {
                  if (Actions.currentScene === 'AgendaDetailViewScene') {
                    Actions.FeedbackFormScene();
                  }
                }}
              >
                <AntDesignIcon name="form" color={Colors.NAV_ICON} size={24} />
              </TouchableOpacity>
            )
          }
        />
        <View style={[styles.timeLineContainer, styles.alignJustifyStart, styles.whiteBackground]}>
          <Text style={[styles.newsTitleText, styles.marginBottom10]}>{eventDetail.title}</Text>
          <Text style={styles.cListLastMessage}>{eventDetail.description}</Text>
        </View>
      </Screen>
    );
  }
}

AgendaDetailView.propTypes = {
  eventDetail: PropTypes.object,
};

AgendaDetailView.defaultProps = {
  eventDetail: {},
};
