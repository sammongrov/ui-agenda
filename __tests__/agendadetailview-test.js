import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Actions } from 'react-native-router-flux';
import AgendaDetailView from '../AgendaDetailView';

configure({ adapter: new Adapter() });

jest.mock('react-native-router-flux', () => ({
  Actions: {
    FeedbackFormScene: jest.fn(),
    pop: jest.fn(),
  },
}));
it('agenda renders correctly', () => {
  const tree = renderer.create(<AgendaDetailView />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Navbar onPress does  calls Actions.pop', () => {
  const root = shallow(<AgendaDetailView />);
  Actions.currentScene = 'AgendaDetailViewScene';
  const button = root
    .find('NavBar')
    .shallow()
    .find('TouchableOpacity')
    .first();
  button.props().onPress();
  expect(Actions.pop.mock.calls.length).toBe(1);
});

it('Navbar onPress does not calls Actions.pop', () => {
  Actions.currentScene = 'Agendascene';
  Actions.pop.mockClear();
  const root = shallow(<AgendaDetailView />);
  const button = root
    .find('NavBar')
    .shallow()
    .find('TouchableOpacity')
    .first();
  button.props().onPress();
  expect(Actions.pop.mock.calls.length).toBe(0);
});

it('Actions.currentScene is true and FeedbackFormScene is called', () => {
  Actions.currentScene = 'AgendaDetailViewScene';
  const rootComp = shallow(<AgendaDetailView />);
  const agendalist = rootComp
    .find('NavBar')
    .first()
    .shallow()
    .find('TouchableOpacity')
    .last();
  agendalist.props().onPress();
  expect(Actions.FeedbackFormScene).toBeCalled();
});

it('Actions.currentScene is false and FeedbackFormScene is not called', () => {
  Actions.currentScene = 'AgendaScene';
  Actions.FeedbackFormScene.mockClear();
  const rootComp = shallow(<AgendaDetailView />);
  const agendalist = rootComp
    .find('NavBar')
    .first()
    .shallow()
    .find('TouchableOpacity')
    .last();
  agendalist.props().onPress();
  expect(Actions.FeedbackFormScene).not.toBeCalled();
});
