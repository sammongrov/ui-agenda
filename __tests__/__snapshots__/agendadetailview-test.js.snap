// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`agenda renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4293914607,
      4293914607,
    ]
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Array [
            undefined,
            null,
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#2E88FF",
                    "fontSize": 36,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
              null,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-Regular",
                },
                Array [
                  undefined,
                  null,
                ],
              ]
            }
          >
            Event Details
          </Text>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "marginRight": 15,
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#2E88FF",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "anticon",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "OpenSans-Regular",
              },
              Array [
                undefined,
                undefined,
              ],
            ]
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "fontFamily": "OpenSans-Regular",
              },
              null,
            ]
          }
        />
      </View>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
